src/stripeClient.ts

```diff
@@ -116,6 +125,42 @@ export class StripeClient {
     return this.cliPath;
   }
 
+  async getOrCreateStripeProcess(
+    stripeProcessName: StripeProcessName,
+    flags: string[] = [],
+  ): Promise<ChildProcess | null> {
+    const existingStripeProcess = this.stripeProcesses.get(stripeProcessName);
+    if (existingStripeProcess) {
+      return existingStripeProcess;
+    }
+
+    const cliPath = await this.getCLIPath();
+    if (!cliPath) {
+      return null;
+    }
+
+    const commandArgs = stripeProcessNameToArgsMap.get(stripeProcessName);
+    if (!commandArgs) {
+      return null;
+    }
+
+    const projectName = vscode.workspace.getConfiguration('stripe').get('projectName', null);
+
+    const allFlags = [...(projectName ? ['--project-name', projectName] : []), ...flags];
+
+    const newStripeProcess = spawn(cliPath, [...commandArgs, ...allFlags]);
+    this.stripeProcesses.set(stripeProcessName, newStripeProcess);
+    return newStripeProcess;
+  }
+
+  endStripeProcess(stripeProcessName: StripeProcessName): void {
```

Comment:
What happens if the child exits on its own, will it ever be removed from this list?

Reply:
These methods don't handle that case, but they should 👍 . Fixing.

Reply:
Added event handlers for `exit` and `error` that will remove the item from the map.

Reply:
So, I think you want to use `close` that's ultimately the event that means the process has completely gone away.

`error` is important in the event it failed to spawn the process as well (or if it failed to kill it).

I am not opposed to you listening to more than one event, but I would hoist all the cleanup logic into a single function that cleaning up. But I suspect you may have more of that in your follow up PRs when you're actually interpreting STDIO.

Reply:
- Changed to `close` instead of `exit`.
- I moved cleanup logic into a separate function, but I don't move the `on` calls. Is that what is meant by "hoist all the cleanup logic into a single function that cleaning up?" (My goal in the end is that whichever class calls `getOrStartCLIProcess` would not have to be concerned about cleanup.)

Reply:
yes, that's what I meant by hoist encapsulating all of that cleanup logic into a common handler.

Generally this looks like directionally what we'd want, I'm sure in follow up PRs which use STDIO I'll have more feedback.

One final nit, in `endCLIProcess` you don't need to explicitly call cleanup, as either `error` or `close` should fire and cleanup would happen there. The "hard" part here is if a caller wants to kill a process and wait for it to quiesce before launching the next one, which will be tricky to handle from a promise/state machine perspective -- as is you remove the reference you're tracking and then can launch a new process (but the old process may still be out there).